src/mbedtls_context.cpp

```diff
@@ -382,6 +382,9 @@ namespace sockpp {
                 case EWOULDBLOCK:
 #if defined(EAGAIN) && EAGAIN != EWOULDBLOCK    // these are usually synonyms
                 case EAGAIN:
+#endif
+#if defined(WSAEWOULDBLOCK)
+				case WSAEWOULDBLOCK:
```

Comment:
This doesn't look right: shouldn't `result.error` be an `errno` value? `WSAEWOULDBLOCK` is a [WSA error code](https://docs.microsoft.com/en-us/windows/win32/winsock/windows-sockets-error-codes-2) returned by [`WSAGetLastError`](https://docs.microsoft.com/en-us/windows/win32/api/winsock/nf-winsock-wsagetlasterror), which is [not the same as an `errno` value](https://docs.microsoft.com/en-us/windows/win32/winsock/error-codes-errno-h-errno-and-wsagetlasterror-2).

Or, if `ioresult::error` is actually a WSA error code on Windows, this switch and other code in this module that uses errno constants (`EPIPE`, `EIO`, …) should also have cases for `WSAEXXX`.

Reply:
If we go along the call stack, bio_return_value is called [here](https://github.com/couchbasedeps/sockpp/blob/8dad5c3c2dc7e5b926bcf4e2fb3ec272dc85dff1/src/mbedtls_context.cpp#L311) with `stream().read_r(buf, length)` which is the errno. The [read_r ](https://github.com/couchbasedeps/sockpp/blob/8dad5c3c2dc7e5b926bcf4e2fb3ec272dc85dff1/src/stream_socket.cpp#L61)function returns ioresult which is ca call to [get_last_error ](https://github.com/couchbasedeps/sockpp/blob/8dad5c3c2dc7e5b926bcf4e2fb3ec272dc85dff1/include/sockpp/stream_socket.h#L73) which is a call to WSAGetLastError which returns WSAEWOULDBLOCK. The get_last_error which fills the error parameter of ioresult is WSAEWOULDBLOCK so result.error is a errno or a WSA error code depending on the OS.

Reply:
Ah, ok. Then I recommend updating the documentation of `ioresult::error` to mention this, and the switch-case here and other places are missing `WSAEXXX` cases for the equivalent of `EPIPE`, `ECONNRESET`, etc.

Reply:
So, yes, I guess it implies that WSA* errors should be added also in other places.

Reply:
I handle this stuff in another class (TCPSocket) by normalizing all WSA errors I know about into POSIX ones but that stuff wouldn't be visible here.  After taking a look, could you change the whitespace to spaces (otherwise the diff looks wonky) and add a similar entry for `WSAECONNRESET` ?  The other POSIX errors here don't appear to have equivalents that I know about.

Reply:
I inserted all equivalent WSA errors that I found.